packages/plugin-betswirl/src/utils/betswirl.ts

```diff
@@ -0,0 +1,303 @@
+import { type Hex } from "viem";
+import { ethers } from "ethers";
+import { MoxieWalletClient } from "@moxie-protocol/moxie-agent-lib/src/wallet";
+import {
+    casinoChainIds,
+    casinoChainById,
+    type CASINO_GAME_TYPE,
+    type CasinoChainId,
+    type GameEncodedInput,
+    type RawBetRequirements,
+    type Token,
+    fetchBetByHash,
+    getBetRequirementsFunctionData,
+    getChainlinkVrfCostFunctionData,
+    getPlaceBetFunctionData,
+    parseRawBetRequirements,
+    chainNativeCurrencyToToken,
+    getAllowanceFunctionData,
+    getApproveFunctionData,
+} from "@betswirl/sdk-core";
+import { getCasinoTokens } from "../providers/casinoTokens";
+
+export async function getChainIdFromWallet() {
+    const chainId = 8453 as CasinoChainId;
+    if (!casinoChainIds.includes(chainId)) {
+        throw new Error(
+            `The chain id must be one of ${casinoChainIds.join(", ")}`
+        );
+    }
+    return chainId;
+}
+
+export async function getBetToken(
+    chainId: CasinoChainId,
+    tokenSymbolInput: string
+) {
+    const casinoChain = casinoChainById[chainId];
+    const casinoTokens = await getCasinoTokens();
+    let selectedToken: Token;
+    if (tokenSymbolInput) {
+        if (
+            tokenSymbolInput.toUpperCase() ===
+            casinoChain.viemChain.nativeCurrency.symbol
+        ) {
+            selectedToken = chainNativeCurrencyToToken(
+                casinoChain.viemChain.nativeCurrency
+            );
+        } else {
+            selectedToken = casinoTokens.find(
+                (casinoToken) =>
+                    casinoToken.symbol === tokenSymbolInput.toUpperCase()
+            );
+        }
+    }
+    if (!selectedToken) {
+        throw new Error(
+            `The token must be one of ${casinoTokens.map((casinoToken) => casinoToken.symbol).join(", ")}`
+        );
+    }
+    return selectedToken;
+}
+
+export function getBetAmountInWei(betAmount: string, token: Token) {
+    const betAmountInWei = ethers.parseUnits(betAmount, token.decimals);
```

Comment:
Let's also do a check above here if there's any `betAmount` and whether it's a valid number, as looks like this currently throw error from ethers library, would be nice to provide more user friendly one
<img width="1087" alt="Screenshot 2025-04-02 at 13 00 57" src="https://github.com/user-attachments/assets/6b82c137-08c7-41c0-bfc5-454fc2795559" />